<?xml version="1.0"?>
<robot xmlns:xacro="http://www.ros.org/wiki/xacro">
    <xacro:macro name="robot_control_control" params="
      name
      prefix
      bus_config
      master_config
      can_interface_name
      master_bin">

        <ros2_control name="${name}" type="system">
            <hardware>
              <plugin>canopen_ros2_control/RobotSystem</plugin>
              <param name="bus_config">${bus_config}</param>
              <param name="master_config">${master_config}</param>
              <param name="can_interface_name">${can_interface_name}</param>
              <param name="master_bin">"${master_bin}"</param>
            </hardware>
            <joint name="${prefix}joint1">
                <param name="device_name">joint_1</param>
                <state_interface name="position" />
            </joint>
            <joint name="${prefix}joint2">
                <param name="device_name">joint_2</param>
                <state_interface name="position" />
            </joint>
        </ros2_control>
    </xacro:macro>
</robot>
